client/sqs/publisher.go

```diff
@@ -6,69 +6,70 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"strconv"
+	"time"
 
-	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
 	"github.com/beatlabs/patron/correlation"
+	"github.com/beatlabs/patron/log"
 	"github.com/beatlabs/patron/trace"
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
-const publisherComponent = "sqs-publisher"
+const (
+	publisherComponent      = "sqs-publisher"
+	attributeDataTypeString = "String"
+)
 
-// Publisher is the interface defining an SQS publisher, used to publish messages to SQS.
-type Publisher interface {
-	Publish(ctx context.Context, msg Message) (messageID string, err error)
+var publishDurationMetrics *prometheus.HistogramVec
+
+func init() {
+	publishDurationMetrics = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "client",
+			Subsystem: "sqs",
+			Name:      "publish_duration_seconds",
+			Help:      "AWS SQS publish completed by the client.",
+		},
+		[]string{"queue", "success"},
+	)
+	prometheus.MustRegister(publishDurationMetrics)
 }
 
-// TracedPublisher is an implementation of the Publisher interface with added distributed tracing capabilities.
-type TracedPublisher struct {
-	api sqsiface.SQSAPI
+type API interface {
```

Comment:
Somebody might argue here that it is not the concern of the publisher to create queues and ReceiveMesages.
The publisher should only care about publishing.
Propose to remove CreateQueue and ReceiveMessage and keep a single method interface.

Reply:
Updated, PTAL. The API interface was used in other testing modules. An interface is present solely for each instance now.